main.js

```diff
@@ -1,117 +1,117 @@
-createLoginForm();
+let validationErrors = {};
+let formValues = {};
+let touchedElements = [];
 
 const userInfo = {
     username: "user",
     password: "test",
     token: ""
 };
 
-const loginForm = document.querySelector("#login")
-const usernameInput = document.querySelector("#username");
-const userPassword = document.querySelector("#password");
-
-loginForm.addEventListener("submit",function(e){
-    
-     e.preventDefault();
-
-    if (isValidUsername() && isValidPassword()) {
-        if (checkData()) {
-            userInfo.token = "xxx"
-            alert("correct");
-        } else{
-            fieldsClear();
-        }
-    } else{
-        fieldsClear();
-    }    
-    
-});
+createLoginForm();
 
 function isValidUsername() {
     const nameRegex = /^[a-zA-Z\-]+$/;
 
-    if(!usernameInput.value){
-        showError("form_username","The username is empty");
-    } else if(usernameInput.value.length > 10 || usernameInput.value.length < 4){
-        showError("form_username","The username should be between 4 - 10");
-    } else if(!usernameInput.value.match(nameRegex)){
-        showError("form_username","the username contains characters that are not allowed");
+    if(!username.value){
```

Comment:
What is `username` here? Are you sure it's working fine?

Reply:
Done